### PR TITLE
[#3232] fix(client): Fix the deleting role issues with irregular name

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Fix the deleting role issues with irregular name.

Add the e2e IT for access control

Fix the issue of null name issue about securable object in the client.

Use NoSuchMetadataObjectException instead of IlegallException for not existed securable object.

### Why are the changes needed?

Fix: #3232

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Add some ITs

### CODE REVIEW

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java [hunk 1]
```diff
@@ -31,44 +31,41 @@
 /** Data transfer object representing a securable object. */
 public class SecurableObjectDTO implements SecurableObject {
 
-  @JsonProperty("fullName")
```
Comment (FANNG1) [inline common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:34]: why changing this class?

**Comment by FANNG1** [comment]
Should we have some constraints on the user&group&role names?

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java (hunk 1 above)
Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:34]: When I added the e2e tests, I find an issue. So I fix it by the way.

**Comment by jerqi** [comment]
> Should we have some constraints on the user&group&role names?

Yes, we can create an issue to track it. But we can discuss further. For this issue, some companies have used the roles with irregular names.So I want to fix this first.

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java (hunk 1 above)
Comment (FANNG1) [reply common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:34]: The PR has mixed fixing, could you add more contexts in `What changes were proposed in this pull request` ?

Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:34]: Done.

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java [hunk 2]
```diff
@@ -31,44 +31,41 @@
...
     if (index == -1) {
-      this.parent = null;
-      this.name = fullName;
+      parent = null;
+      name = fullName;
     } else {
-      this.parent = fullName.substring(0, index);
-      this.name = fullName.substring(index + 1);
+      parent = fullName.substring(0, index);
+      name = fullName.substring(index + 1);
```
Comment (jerryshao) [inline common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:59]: Why do you remove `this.` in here?

Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:59]: I removed the constructor method. This is method `setFullName`.

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java [hunk 3]
```diff
@@ -31,44 +31,41 @@
...
     }
-
-    this.privileges = privileges;
   }
 
+  private String parent;
+  private String name;
+
+  /** Default constructor for Jackson deserialization. */
+  protected SecurableObjectDTO() {}
```
Comment (jerryshao) [inline common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:67]: Why don't you move these lines to the top of the class?

File: integration-test/src/test/java/org/apache/gravitino/integration/test/authorization/AccessControlIT.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    String username = "user1#123";
+    User user = metalake.addUser(username);
+    Assertions.assertTrue(user.roles().isEmpty());
+
+    String roleName = "role#123";
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("k1", "v1");
+    SecurableObject metalakeObject =
+        SecurableObjects.ofMetalake(
+            metalakeName, Lists.newArrayList(Privileges.CreateCatalog.allow()));
```
Comment (jerryshao) [inline integration-test/src/test/java/org/apache/gravitino/integration/test/authorization/AccessControlIT.java:162]: Will you check if the metadata object exist or not? If so, please add the test to cover this.

File: common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java (hunk 3 above)
Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/authorization/SecurableObjectDTO.java:67]: I will.

File: integration-test/src/test/java/org/apache/gravitino/integration/test/authorization/AccessControlIT.java (hunk 4 above)
Comment (jerqi) [reply integration-test/src/test/java/org/apache/gravitino/integration/test/authorization/AccessControlIT.java:172]: Added the test case.

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java [hunk 5]
```diff
@@ -242,6 +242,7 @@ public boolean deleteRole(String role) throws NoSuchMetalakeException {
    * @return The created Role instance.
    * @throws RoleAlreadyExistsException If a Role with the same name already exists.
    * @throws NoSuchMetalakeException If the Metalake with the given name does not exist.
+   * @throws NoSuchMetadataObjectException If securable object doesn't exist
    * @throws RuntimeException If creating the Role encounters storage issues.
    */
   public Role createRole(
```
Comment (jerryshao) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:248]: Should you also add this exception to the interface signature?

Comment (jerqi) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:248]: Added.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**